unity-ggjj/Assets/Tests/PlayModeTests/Tools/StoryProgresser.cs

```diff
@@ -15,5 +19,82 @@ public IEnumerator ProgressStory()
             yield return TestTools.WaitForState(() => !appearingDialogueController.IsPrintingText);
             Release(Keyboard.xKey);
         }
+
+        /// <summary>
+        /// Selects a choice and presses the x key to progress the story
+        /// </summary>
+        /// <param name="choiceIndex">The index of the choice to select</param>
+        /// <param name="gameMode">The GameMode the game is currently using</param>
+        /// <param name="evidenceName">The name of any evidence to present (if any)</param>
+        public IEnumerator SelectChoice(int choiceIndex, GameMode gameMode, EvidenceAssetName evidenceName)
+        {
+            switch (gameMode)
+            {
+                case GameMode.Dialogue:
+                    var choice = Object.FindObjectOfType<ChoiceMenu>().transform.GetChild(choiceIndex).GetComponent<Selectable>();
+                    choice.Select();
+                    yield return new WaitForSeconds(.75f);
+                    yield return PressForFrame(Keyboard.xKey);
+                    break;
+                case GameMode.CrossExamination:
+                    var evidenceMenu = Object.FindObjectOfType<EvidenceMenu>();
+                    if (evidenceMenu != null && evidenceMenu.CanPresentEvidence)
+                    {
+                        var narrativeGameState = Object.FindObjectOfType<NarrativeGameState>();
+                        var courtRecordObjects = narrativeGameState.ObjectStorage.GetObjectsOfType<ICourtRecordObject>().ToList();
+
+                        if (courtRecordObjects.Any(courtRecordObject => courtRecordObject is ActorData && courtRecordObject.InstanceName == evidenceName))
+                        {
+                            yield return PressForFrame(Keyboard.cKey);
+                        }
+                        else if (courtRecordObjects.All(courtRecordObject => courtRecordObject.InstanceName != evidenceName))
```

Comment:
Nit: These LINQ expressions are quite a mouthful! : D  
I think this is one of the few cases, where I would suggest decorating lines with comments, as I also don't see any way of expressing them differently in code:
```suggestion
                        // when trying to select an actor as evidence, we first need to flip to the Actor Court record page
                        if (courtRecordObjects.Any(courtRecordObject => courtRecordObject is ActorData && courtRecordObject.InstanceName == evidenceName))
                        {
                            yield return PressForFrame(Keyboard.cKey);
                        }
                        // if none of the court records we have match the evidenceName, we exit this coroutine
                        else if (courtRecordObjects.All(courtRecordObject => courtRecordObject.InstanceName != evidenceName))
```